CMakeLists.txt

```diff
@@ -49,6 +49,8 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 set(SONATA_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include)
 
+set(SONATA_COMPILE_OPTIONS "${SONATA_COMPILE_OPTIONS}" "-Wp,-D_GLIBCXX_ASSERTIONS")
```

Comment:
not a huge fan of using glibcxx specific options (ie: things that libc++ doesn't support).

However, I do like catching these sorts of things, so I'm torn.  At the minimum, I think we should only enable it for `SONATA_CXX_WARNINGS`, and gate it by the c++ library being used, which I will have to look into.

We'll take it from here; thanks @sanjayankur31 very much, this is excellent feedback.

Reply:
Thanks very much @mgeplf . We'll have libsonata included in Fedora soon and then it'll be built for all the architectures we support. I'll report anything else that we catch too with PRs where possible.